### PR TITLE
working PoC fastify migration

### PR DESCRIPTION
**Pull request by eadmundo**
As per https://twitter.com/kentcdodds/status/1680977526204559361?s=46&t=Zmcf63TwFKrCwhRSBkdZlg

This is a mostly working migration off express to fastify.

## TODO:

 - [x] restore `app/routes/settings+/profile.photo.tsx` without `File` causing reference error
 - [x] plugin for trailing slash redirect
 - [x] plugin for HTTPS only
 - [ ] ~port `@sentry/remix` request handler wrapper~ switch to fastify sentry plugin or open-telemetry
 - [ ] switch between devBuild/build in request handler
 - [x] ~migrate `closeWithGrace`~ builtin to fastify
 - [x] ~replicate `morgan` tiny format logging in `pino`~ not sure this is necessary
 - [x] refactor, maybe to use autoload
 - [ ] clean-up and remove unneeded dependencies

### CODE REVIEW

File: app/routes/settings+/profile.photo.tsx [hunk 1]
```diff
@@ -30,12 +30,14 @@ export const handle = {
...
-			return file.size <= MAX_SIZE
-		}, 'Image size must be less than 3MB'),
+	photoFile: typeof window === 'undefined'
+		? z.undefined()
+		: z
+			.instanceof(File)
+			.refine(file => file.name !== '' && file.size !== 0, 'Image is required')
+			.refine(file => {
+				return file.size <= MAX_SIZE
+			}, 'Image size must be less than 3MB'),
```
Comment (kentcdodds) [inline app/routes/settings+/profile.photo.tsx:40]: We definitely would still want this validation on the server. You may need to use Remix's `installGlobals`: https://github.com/remix-run/remix/blob/f2c4edfc28f7427189e67f32f12628e1c45b6b69/packages/remix-node/globals.ts#L45

**Review by kentcdodds — COMMENTED** [review]
This is looking awesome! I'm not sure I'm ready to merge just yet, but would you be willing to put together an official example? https://github.com/epicweb-dev/epic-stack/blob/main/docs/examples.md

**Comment by kentcdodds** [comment]
Thanks for working on this. I'm going to close due to lack of activity. Feel free to open again if you'd like to finish and propose the change. You could also make an example for the docs

**Comment by eadmundo** [comment]
Thanks @kentcdodds - I've got an example pretty much ready to go [here](https://github.com/autotelic/epic-stack-with-fastify), and I'll do a PR for the docs as soon as it's complete!

**Comment by kentcdodds** [comment]
Stellar! Looking forward to it :)

**Pull request closed without merging**